src/linechart/DataSeries.jsx

```diff
@@ -12,129 +11,90 @@ module.exports = React.createClass({
+    colorAccessor: React.PropTypes.func,
+    interpolationType: React.PropTypes.string
   },
 
   getDefaultProps() {
     return {
       data: [],
-      seriesName: "",
-      interpolationType: 'linear',
       fill: '#fff',
       xAccessor: (d) => d.x,
       yAccessor: (d) => d.y,
-      displayDataPoints: true
+      interpolationType: 'linear'
     };
   },
-
+  
   _isDate(d, accessor) {
-    return Object.prototype.toString.call(accessor(d)) === '[object Date]';
+      return Object.prototype.toString.call(accessor(d)) === '[object Date]';
   },
 
   render() {
-
     var props = this.props;
-
-    // Check to see if there is any data to draw lines/circles with
-    // Return null if there is nothing to draw (null will become <noscript>)
-    if (props.data.length === 0) {
-      return null;
-    }
-
+    var xScale = props.xScale;
+    var yScale = props.yScale;
     var xAccessor = props.xAccessor,
-        yAccessor = props.yAccessor;
-
-    // Create array of paths, which we'll map over
-    // to generate SVG lines
+        yAccessor = props.yAccessor,
+        cx, cy, circleFill;
+    
     var interpolatePath = d3.svg.line()
-        .y(function(d) {
-          return props.yScale(props.yAccessor(d));
-        })
+        .y( (d) => props.yScale(yAccessor(d)) )
         .interpolate(props.interpolationType);
 
-    // Check whether or not an arbitrary data element
-    // is a date object (at index 0 here)
-    // If it's a date, then we set the x scale a bit differently
-    // NOTE: a sparse array could still break this, even with the length check above
-    if (this._isDate(props.data[0], xAccessor)) {
-      interpolatePath.x(function(d) {
-        return props.xScale(props.xAccessor(d).getTime());
-      });
-    } else {
-      interpolatePath.x(function(d) {
-        return props.xScale(props.xAccessor(d));
-      });
-    }
-
-    // Create an immstruct reference for the series name
-    // and set it to 'inactive'
-    props.structure.cursor('voronoiSeries').set(props.seriesName, 'inactive');
-
-    // Having set the Voronoi line series name cursor to 'inactive'
-    // We now pass on the Voronoi line series name reference to the
-    // *both* the line and circle component
-    var voronoiSeriesRef = props.structure.reference(['voronoiSeries', props.seriesName]);
-
-
-    var circles = null;
-
-    if (props.displayDataPoints) {
-      // Map over data to generate SVG circles at data points
-      // if datum is a date object, treat it a bit differently
-      circles = props.data.map(function(point, idx) {
-        var cx, cy;
-        if (this._isDate(point, xAccessor)) {
-          cx = props.xScale(xAccessor(point).getTime());
-        } else {
-          cx = props.xScale(xAccessor(point));
-        }
-        if (this._isDate(point, yAccessor)) {
-          cy = props.yScale(yAccessor(point).getTime());
+        if (this._isDate(props.data[0].values[0], xAccessor)) {
```

Comment:
Since you removed the check above

```javascript
// Check to see if there is any data to draw lines/circles with
// Return null if there is nothing to draw (null will become <noscript>)
if (props.data.length === 0) {
     return null;
}

this is now going to cause a breaking nullPointerException if `props.data` is empty, which it is by default...

(You need to pay more attention to what you are removing.)